Revit_Engine/Compute/RevitDiffing.cs

```diff
@@ -62,150 +85,87 @@ public static Diff RevitDiffing(IEnumerable<object> pastObjects, IEnumerable<obj
         [Input("propertiesOrParamsToConsider", "Object properties to be considered when comparing two objects for differences. If null or empty, all properties will be considered." +
         "\nYou can specify also Revit parameter names.")]
         [Input("diffConfig", "Further Diffing configurations.")]
         [Output("diff", "Holds the differences between the two sets of objects. Explode it to see all differences.")]
-        private static Diff Diffing(IEnumerable<object> pastObjects, IEnumerable<object> followingObjects, string revitIdName = "UniqueId", IEnumerable<string> propertiesOrParamsToConsider = null, DiffingConfig diffConfig = null)
+        private static Diff Diffing(IEnumerable<object> pastObjects, IEnumerable<object> followingObjects, string revitIdName = "UniqueId", DiffingConfig diffConfig = null)
         {
-            // Set configurations if diffConfig is null. Clone it for immutability in the UI.
-            DiffingConfig diffConfigClone = diffConfig == null ? new DiffingConfig() { IncludeUnchangedObjects = true } : diffConfig.DeepClone();
-
-            if (pastObjects == null)
-                pastObjects = new List<object>();
-
-            if (followingObjects == null)
-                followingObjects = new List<object>();
-
-            // // - Now do the necessary configurations for the Revit-specific diffing.
+            // Checks and setup of input objects.
+            if (pastObjects == null) pastObjects = new List<object>();
+            if (followingObjects == null) followingObjects = new List<object>();
 
             // Checks and setup of revitIdName.
-            if (revitIdName == "UniqueId" || revitIdName == nameof(RevitIdentifiers.PersistentId))
+            if (revitIdName == "UniqueId" || revitIdName == nameof(RevitIdentifiers.PersistentId) || revitIdName.IsNullOrEmpty())
                 revitIdName = nameof(RevitIdentifiers.PersistentId);
             else if (revitIdName != "ElementId")
             {
                 BH.Engine.Reflection.Compute.RecordError($"The input parameter {nameof(revitIdName)} can only be 'ElementId', 'UniqueId' or '{nameof(RevitIdentifiers.PersistentId)}' (BHoM's equivalent of Revit's UniqueId), but '{revitIdName}' was specified.");
                 return null;
             }
 
-            // Get only the BHoMObjects from the input collections.
-            IEnumerable<IBHoMObject> pastBHoMObjects = pastObjects.OfType<IBHoMObject>();
-            IEnumerable<IBHoMObject> followingBHoMObjects = followingObjects.OfType<IBHoMObject>();
+            // Checks and setup of DiffingConfig/ComparisonConfig.
+            DiffingConfig diffConfigClone = diffConfig == null ? new DiffingConfig() { IncludeUnchangedObjects = true } : diffConfig.DeepClone();
+            RevitComparisonConfig rcc = diffConfigClone.ComparisonConfig as RevitComparisonConfig;
+            if (rcc == null)
+            {
```

Comment:
Isn't that worth a warning/note? Not sure if this is helpful to the user by any means, so please treat this comment as a question.

Reply:
I can add a warning

Reply:
No sorry actually it does not make sense to add a warning. The input is always a DiffingConfig. If the DiffingConfig is also a RevitComparisonConfig (child of DiffingConfig) then good, otherwise just instantiate a RevitComparisonConfig and copy over the properties from the input DiffingConfig.